### PR TITLE
Js send json

### PR DESCRIPTION
**Pull request by jsingle**
@thieman this should fix the security vulnerability.  I like this system of just using a decorator to register a function, instead of typing out a big map somewhere

### CODE REVIEW

File: tests/integration/cli/config_test.py [hunk 1]
```diff
@@ -27,4 +27,4 @@ def test_config_set_fails_on_nonexistent_key(self):
     def test_config_set_works_with_valid_input(self):
         self.run_command('config set nginx_includes_dir /var/nginx')
         result = yaml.load(self.run_command('config listvalues'))
-        self.assertEqual(result['nginx_includes_dir'], '/var/nginx')
+        self.assertEqual(result['nginx_includes_dir'], u'/var/nginx')
```
Comment (thieman) [inline tests/integration/cli/config_test.py]: This shouldn't be necessary, was it failing without it?

File: dusty/payload.py [hunk 2]
```diff
@@ -20,11 +20,30 @@ def run(self):
...
+            raise RuntimeError('Function key {} not found; you may need to decorate your function'.format(fn_key))
+        doc = {'fn_key': fn_key, 'client_version': self.client_version, 'suppress_warnings': self.suppress_warnings,
                'args': self.args, 'kwargs': self.kwargs}
-        return cPickle.dumps(doc).encode('string_escape')
+        return json.dumps(doc).encode('string_escape')
 
     @staticmethod
     def deserialize(doc):
-        return cPickle.loads(doc.decode('string_escape'))
+        return json.loads(doc.decode('string_escape'))
```
Comment (thieman) [inline dusty/payload.py]: Do we still need `string_escape` if we're doing JSON?

File: dusty/payload.py [hunk 3]
```diff
@@ -20,11 +20,30 @@ def run(self):
...
 
+_daemon_command_mapping = {}
+
+def _get_key(f):
+    return '{}.{}'.format(f.__module__, f.__name__)
```
Comment (thieman) [inline dusty/payload.py]: This function needs a better name

File: dusty/payload.py [hunk 4]
```diff
@@ -20,11 +20,30 @@ def run(self):
...
+
+def daemon_command(f):
+    key = _get_key(f)
+    if key in _daemon_command_mapping and _daemon_command_mapping[key] != f:
+        raise RuntimeError("Function mapping key collision: {}. Name one of the functions something else".format(key))
+    _daemon_command_mapping[key] = f
+    return f
```
Comment (thieman) [inline dusty/payload.py]: `fn` would be a little better for comprehension than `f`

File: tests/integration/cli/config_test.py (hunk 1 above)
Comment (jsingle) [inline tests/integration/cli/config_test.py]: Yeah, and with it.  The issue is that json loads makes all the strings unicode, even if you dumped non-unicode strings.  Apparently using yaml.safe_load gets around this (but won't work the same for weird character sets)

**Comment by thieman** [comment]
Like the design, couple nitpicks otherwise :bear:

**Comment by jsingle** [comment]
Still gotta look into string escape.  I wish I understood how strings work :/

**Pull request merged**